### PR TITLE
PMM-6693 Add support of API keys.

### PR DESCRIPTION
**Pull request by BupycHuk**
[PMM-6693](https://jira.percona.com/browse/PMM-6693)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1479

- [ ] Links to other linked pull requests (optional).
- https://github.com/percona/pmm-server-packaging/pull/204
- https://github.com/percona-platform/grafana/pull/61
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Review by askomorokhov — APPROVED** [review]

**Comment by askomorokhov** [comment]
@BupycHuk, can we cover this with API tests?

File: services/grafana/client.go [hunk 1]
```diff
@@ -181,13 +181,19 @@ func (r role) String() string {
 func (c *Client) getRole(ctx context.Context, authHeaders http.Header) (role, error) {
 	// https://grafana.com/docs/http_api/user/#actual-user - works with any authentication
 	var m map[string]interface{}
-	if err := c.do(ctx, "GET", "/api/user", "", authHeaders, nil, &m); err == nil {
-		if a, _ := m["isGrafanaAdmin"].(bool); a {
-			return grafanaAdmin, nil
+	err := c.do(ctx, "GET", "/api/user", "", authHeaders, nil, &m)
+	// User is not exist, check API key role.
+	if err != nil {
+		if e, ok := errors.Cause(err).(*clientError); ok && e.Code == 404 {
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply services/grafana/client.go:187]: I tried to keep previous behavior for basic auth. Maybe you are right. Will fix it.

**Review by artemgavrilov — APPROVED** [review]

File: services/grafana/client.go [hunk 2]
```diff
@@ -202,22 +214,60 @@ func (c *Client) getRole(ctx context.Context, authHeaders http.Header) (role, er
...
+	case "Editor":
+		return editor
+	case "Admin":
+		return admin
+	default:
+		return none
+	}
+}
+
+func (c *Client) getRoleforAPIKey(ctx context.Context, authHeaders http.Header) (role, error) {
```
Comment (atymchuk) [inline services/grafana/client.go:252]:
```suggestion
func (c *Client) getRoleForAPIKey(ctx context.Context, authHeaders http.Header) (role, error) {
```

File: services/grafana/client.go [hunk 3]
```diff
@@ -202,22 +214,60 @@ func (c *Client) getRole(ctx context.Context, authHeaders http.Header) (role, er
...
+	var o map[string]interface{}
+	if err := c.do(ctx, "GET", "/api/org", "", authHeaders, nil, &o); err != nil {
```
Comment (atymchuk) [inline services/grafana/client.go:254]: Now that we have a dedicated endpoint `/api/auth/key`, we don't need to make an extra call to `/api/org` (which felt like a code smell anyway).  So, I'd suggest we skip lines 253 trough 260.

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: services/grafana/client.go [hunk 4]
```diff
@@ -202,22 +211,55 @@ func (c *Client) getRole(ctx context.Context, authHeaders http.Header) (role, er
...
+	}
+}
+
+func (c *Client) getRoleForAPIKey(ctx context.Context, authHeaders http.Header) (role, error) {
+	var k map[string]interface{}
+	if err := c.do(ctx, "GET", "/api/auth/key", "", authHeaders, nil, &k); err != nil {
+		return none, err
+	}
+
+	if id, _ := k["orgId"].(float64); id != 1 {
```
Comment (atymchuk) [inline services/grafana/client.go:255]: What if the Customers use multiple orgs?

Comment (BupycHuk) [reply services/grafana/client.go:255]: Now we don't support multiple orgs.
For users we support only default org, so I decided that we should use the same behavior for API keys.

Comment (BupycHuk) [reply services/grafana/client.go:255]: https://github.com/percona/pmm-managed/pull/653/files#diff-a9c8e70f3cab071e6bbcb1cc40e5c8a0f5d9bc87cdc48b0fca4e3c70466d12d6R212

**Review by atymchuk — APPROVED** [review]

**Comment by BupycHuk** [comment]
Failed tests are expected because I've made changes in grafana, but pmm-server:dev-latest is used in tests

**Pull request merged**